### ISSUE TITLE
Entry signals not executed: only exits (sell/close_short) is triggered for long

### ISSUE THREAD

**Issue by adenm1** [freqtrade/freqtrade#12049]
Environment
	•	Freqtrade version: v2025.7 (latest from GitHub)
	•	Python version: 3.13.5 (.venv)
	•	Operating System: macOS
	•	Run mode: dry_run (futures)
	•	Exchange: Binance Futures (futures_mode = true, margin_mode = isolated)

Run the bot in dry-run: freqtrade trade --strategy AdaptiveStrategy --config config.json --dry-run
What Happened
	•	Only Long entries (entry_long) occur, even when the MLP prediction is positive and 1h trend is up.
	•	Short entries (entry_short) occasionally log [DEBUG SHORT EMIT], but no actual positions open.
	•	Only exit signals (exit_long / exit_short triggered by stop-loss or take-profit) ever execute, causing every trade to exit by stop-loss.

Relevant log excerpt:
`[DEBUG SHORT ENTRY] BTC/USDT:USDT latest idx=1499
2025-07-30 22:14:40,007 - AdaptiveStrategy - INFO - MLP predicted return for short BTC/USDT:USDT at 1499: 0.45%
[DEBUG SHORT MLP] BTC/USDT:USDT at 1499: mlp_signal=False, ml_pred=0.004470
RL short action for BTC/USDT:USDT at 1499: [ True ]
[DEBUG SHORT EMIT] Emitting entry_short signal for BTC/USDT:USDT at 1499, mlp_signal=False, rl_signal=True
[DEBUG SELL EMIT] BTC/USDT:USDT at 101: signal=SELL, ml_pred=-0.007694, entry_price=115877.2, last_close=115989.9
...`

Strategy.py:
```
class AdaptiveStrategy(IStrategy):
    INTERFACE_VERSION = 3
    can_short = True
    trading_mode = "futures"
    margin_mode = "isolated"
    timeframe = "5m"
    stoploss = -0.04
    LOOKBACK = 60
    ...

    def populate_buy_trend(self, dataframe, metadata):
        dataframe['entry_long'] = 0
        # … load models …
        last_buy = self._last_buy_bar.get(metadata['pair'])
        in_position = last_buy is not None
        for i in range(self.LOOKBACK, len(dataframe)):
            ml_pred = mlp_model.predict(...)[-1]
            if ml_pred > 0.004 and not in_position and dataframe['in_1h_uptrend'].iat[i]:
                dataframe.loc[dataframe.index[i], 'entry_long'] = 1
                in_position = True
        return dataframe

    def populate_short_trend(self, dataframe, metadata):
        print(f"[DEBUG SHORT ENTRY] {metadata['pair']} latest idx={dataframe.index[-1]}")
        dataframe['entry_short'] = 0
        # … load models, preprocess …
        mlp_signal = preds_mlp[-1] < -0.004
        rl_signal = (action == 2)
        if mlp_signal or rl_signal:
            dataframe.loc[df_clean.index[-1], 'entry_short'] = 1
            print(f"[DEBUG SHORT EMIT] Emitting entry_short for {metadata['pair']} at {df_clean.index[-1]}, "
                  f"mlp_signal={mlp_signal}, rl_signal={rl_signal}")
        return dataframe
```

Configuration Snippet (config.json)
```
{ 
  "dry_run": true,
  "futures_mode": true,
  "margin_mode": "isolated",
  "shorts_enabled": true,
  "pair_whitelist": [
    "BTC/USDT:USDT", "ETH/USDT:USDT", "SUI/USDT:USDT", ...
  ],
  "ccxt_config": {
    "enableRateLimit": true,
    "options": { "defaultType": "future" }
  },
  "exchange": {
    "name": "binance",
    ...
  },
  "use_entry_signal": true,
  "use_exit_signal": true,
  ...
```

What I've Tried
	•	Verified INTERFACE_VERSION = 3, can_short = True.
	•	Confirmed 1h trend filter is correctly aligned to 5m dataframe.
	•	Observed debug logs showing entry conditions met, yet no actual positions open.
	•	Tried both entry_long/entry_short and buy/short column names.
Any pointers on what might be preventing entry_long or entry_short from actually being recognized and executed? Thanks in advance!

**Comment by froggleston** [comment]
Your functions are named incorrectly. Please follow the quick start: https://www.freqtrade.io/en/stable/strategy-101/#a-simple-strategy

`populate_entry_trend`
`populate_exit_trend`
`enter_long` vs `enter_short`
`exit_long` vs `exit_short`

`entry_long` and `entry_short` are incorrect, as is `populate_short_trend`

**Comment by adenm1** [comment]
> Your functions are named incorrectly. Please follow the quick start: https://www.freqtrade.io/en/stable/strategy-101/#a-simple-strategy
> 
> `populate_entry_trend` `populate_exit_trend` `enter_long` vs `enter_short` `exit_long` vs `exit_short`
> 
> `entry_long` and `entry_short` are incorrect, as is `populate_short_trend`

Thanks.
Is it ok that I wrote these code?:
```    
    def populate_entry_trend(self, dataframe: pd.DataFrame, metadata: dict) -> pd.DataFrame:
        df = self.populate_buy_trend(dataframe, metadata)
        df = self.populate_short_trend(df, metadata)
        return df

    def populate_exit_trend(self, dataframe: pd.DataFrame, metadata: dict) -> pd.DataFrame:
        df = self.populate_sell_trend(dataframe, metadata)
        df = self.populate_close_short_trend(df, metadata)
        return df
```

**Comment by froggleston** [comment]
After fixing `entry_short` etc.

**Comment by adenm1** [comment]
> After fixing `entry_short` etc.

Thank you very much!!!! it's working now!!!!

**Comment by froggleston** [comment]
Even though you mentioned dry run, please bear in mind the code you've written there isn't backtestable.

**Comment by adenm1** [comment]
> Even though you mentioned dry run, please bear in mind the code you've written there isn't backtestable.

Thanks!
